tests/migrations/test_optimizer.py

```diff
@@ -270,6 +270,35 @@ def test_optimize_through_create(self):
             app_label="testapp",
         )
 
+        # This could be optimized a bit more but at least it generates a valid
+        # set of operations.
+        self.assertOptimizesTo(
+            [
+                migrations.CreateModel('Book', [('name', models.CharField(max_length=255))]),
+                migrations.CreateModel('Person', [('name', models.CharField(max_length=255))]),
+                migrations.AddField('book', 'author', models.ForeignKey('test_app.Person', models.CASCADE)),
+                migrations.CreateModel('Review', [('book', models.ForeignKey('test_app.Book', models.CASCADE))]),
+                migrations.CreateModel('Reviewer', [('name', models.CharField(max_length=255))]),
+                migrations.AddField('review', 'reviewer', models.ForeignKey('test_app.Reviewer', models.CASCADE)),
+                migrations.RemoveField('book', 'author'),
+                migrations.DeleteModel('Person'),
+            ],
+            [
+                migrations.CreateModel('Person', [('name', models.CharField(max_length=255))]),
+                migrations.CreateModel('Book', [
+                    ('name', models.CharField(max_length=255)),
+                    ('author', models.ForeignKey('test_app.Person', models.CASCADE)),
+                ]),
+                migrations.CreateModel('Reviewer', [('name', models.CharField(max_length=255))]),
+                migrations.CreateModel('Review', [
+                    ('book', models.ForeignKey('test_app.Book', models.CASCADE)),
+                    ('reviewer', models.ForeignKey('test_app.Reviewer', models.CASCADE)),
+                ]),
+                migrations.RemoveField('book', 'author'),
```

Comment:
Looking through this, it might not actually matter that the field might be a foreign key here.

Because this operation is "remove this field from the model" and not an alter/add, you don't need any information about the field to execute on this. Just like you don't need information to run a "drop column". Basically, I don't think the `RemoveField` operation references the model it points to, in the case where the original field is a foreign key.

I do not have a mathematical proof for this but I feel like the fact that the migration executor doesn't need to have details of the field is decent proof.